src/ffi.rs

```diff
@@ -841,3 +871,36 @@ pub extern fn quiche_conn_dgram_purge_outgoing(
 pub extern fn quiche_conn_free(conn: *mut Connection) {
     unsafe { Box::from_raw(conn) };
 }
+
+#[cfg(unix)]
+fn instant_to_timespec(instant: std::time::Instant) -> timespec {
+    use std::hash::{
+        Hash,
+        Hasher,
+    };
+
+    struct ToTimeSpec(Vec<i64>);
+    impl Hasher for ToTimeSpec {
+        fn finish(&self) -> u64 {
+            unimplemented!();
+        }
+
+        fn write(&mut self, _: &[u8]) {
+            unimplemented!();
+        }
+
+        fn write_u64(&mut self, i: u64) {
+            self.0.push(i as _);
+        }
+
+        fn write_u32(&mut self, i: u32) {
+            self.0.push(i as _);
+        }
+    }
+    let mut hasher = ToTimeSpec(vec![]);
+    instant.hash(&mut hasher);
```

Comment:
TBH, I kind of prefer the `std::ptr::copy_nonoverlapping()` alternative to this, as it is clearer what it does, and is also less code.

Reply:
thought hash,hasher method was safe and predictable but anyways, will go with copy_nonoverlapping as it is clearer.